src/braket/default_simulator/simulator.py

```diff
@@ -264,8 +265,12 @@ def _create_results_obj(
                 action=openqasm_ir,
             ),
             resultTypes=results,
-            measurements=self._formatted_measurements(simulation),
-            measuredQubits=self._get_measured_qubits(simulation.qubit_count),
+            measurements=self._formatted_measurements(simulation, measured_qubits),
+            measuredQubits=(
+                measured_qubits
+                if measured_qubits
+                else self._get_measured_qubits(simulation.qubit_count)
```

Comment:
this method clearly doesn't fulfill its name anymore. Fortunately, it's a private method, so we can refactor the code here

Reply:
I renamed this method to `_get_all_qubits` as that seems more accurate.